src/components/custom/review/review.tsx

```diff
@@ -32,20 +39,139 @@ export const Review = (props: IGenericCustomElementProps<IReviewSchema>) => {
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, []);
 
+	useEffect(() => {
```

Comment:
side note: potentially resets the failure count if the schema changes and causes the `schema.items` reference to change. tested with a static schema and didn't trigger the useEffect, so I don't think it's an actual issue

Reply:
just to be safe, I changed to use `useDeepCompareEffect` instead so `itemDetailList` will only be updated if the items have truly changed.